wandb/cli/cli.py

```diff
@@ -1422,15 +1422,8 @@ def local(ctx, port, env, daemon, upgrade, edge):
     api = InternalApi()
     if not find_executable("docker"):
         raise ClickException("Docker not installed, install it from https://docker.com")
-    if wandb.docker.image_id("wandb/local") != wandb.docker.image_id_from_registry(
-        "wandb/local"
-    ):
-        if upgrade:
-            subprocess.call(["docker", "pull", "wandb/local"])
-        else:
-            wandb.termlog(
-                "A new version of W&B local is available, upgrade by calling `wandb local --upgrade`"
-            )
+    if upgrade:
+        subprocess.call(["docker", "pull", "wandb/local"])
```

Comment:
I think we still need to inform the user an upgrade is available / educate them that `wandb local --upgrade` exists...  I saw a few bug reports about the `image_id_from_registry` returning an error, but we should fix that underlying error or change this check to detect a new version using the api endpoint...

Reply:
In this specific case the issue was that the `image_id_from_registry` would also include the registry address. i.e:
```
>> print(wandb.docker.image_id("wandb/local"))
'wandb/local@sha256:6edf712e7b7c62383ab78d50c77321f3ac301b9a83fe3c130f5e1733c76ace65'

>> print(wandb.docker.image_id_from_registry("wandb/local"))
'registry-1.docker.io/wandb/local@sha256:6edf712e7b7c62383ab78d50c77321f3ac301b9a83fe3c130f5e1733c76ace65'
```
so to fix this I could just take the final hash and compare...